### DISCUSSION TITLE
share connection handler between multiple namespace

### DISCUSSION THREAD

**Discussion by nocountryforusreal** [Totodore/socketioxide#226] [General]
I want to add namespace dynamically at run time with same connection handler for default namespace. So how to sharing handler between many ns

**Comment by tausifcreates** [comment]
Hi. Dynamic Namespaces is not yet implemented. See https://github.com/Totodore/socketioxide/issues/212

**Reply by Totodore** [reply]
As an alternative you can put a middleware in front of socketio and dynamically declare new namespaces if needed and give the same `on_connect` handler.

**Reply by tausifcreates** [reply]
Should we mark your answer as resolved for now?

**Reply by nocountryforusreal** [reply]
middleware won't work after protocol upgraded

**Reply by Totodore** [reply]
You're right, you will have to wait for namespace routing.

**Comment by nocountryforusreal** [comment]
hi, has [Dynamic Namespaces] been added to recently release?

**Reply by Totodore** [reply]
No, #212 is still blocked because we are waiting for https://github.com/ibraheemdev/matchit/pull/49

**Reply by nocountryforusreal** [reply]
thanks

**Reply by Totodore** [reply]
Hi @nocountryforusreal, new release v0.14.0 introduce dynamic namespaces.
